File: src/business/rule/engine/serviceprovider/DefaultBusinessRuleProvider.java
package business.rule.engine.serviceprovider;

import business.rule.engine.model.Order;
import business.rule.engine.service.IUserService;

public class DefaultBusinessRuleProvider implements IBusinessRuleProvider
    {
        private IUserService userService;

        public DefaultBusinessRuleProvider(IUserService userService)
        {
            this.userService = userService;
        }

        public String NameRequirement = String.Empty;

        public boolean isSatisfied Process(Order order)
        {
            return true;
        }
    }

File: src/BusinessRuleProviders/DefaultBusinessRuleProvider.cs
using Raymaker.RulesEngine.Application.Model;

namespace Raymaker.RulesEngine.Application.BusinessRuleProviders
{
    public class DefaultBusinessRuleProvider : IBusinessRuleProvider
    {
        private readonly IUserService userService;

        public DefaultBusinessRuleProvider(IUserService userService)
        {
            this.userService = userService;
        }

        public string NameRequirement => string.Empty;

        public (bool isSatisfied, string message) Process(Order order)
        {
            return (true, $"{NameRequirement}: No action");
        }
    }
}
